builtin/providers/clc/resource_clc_group.go

```diff
@@ -0,0 +1,118 @@
+package clc
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/CenturyLinkCloud/clc-sdk"
+	"github.com/CenturyLinkCloud/clc-sdk/group"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCLCGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCLCGroupCreate,
+		Read:   resourceCLCGroupRead,
+		Update: resourceCLCGroupUpdate,
+		Delete: resourceCLCGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"location_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"custom_fields": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+		},
+	}
+}
+
+func resourceCLCGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("location_id").(string)
+	m, err := dcGroups(dc, meta)
+	if err != nil {
+		return fmt.Errorf("Failed pulling groups in location %v - %v", dc, err)
+	}
+	name := d.Get("name").(string)
+	// use an existing group if we have one
+	if m[name] != "" {
+		log.Printf("[INFO] Using EXISTING group: %v => %v", name, m[name])
+		d.SetId(m[name])
+		return nil
+	}
```

Comment:
Ah yes - thorny indeed!

I do see the merits in argument you're making, but the fact that groups don't have any uniqueness constraint makes me worry. What would the behavior be when there are two groups named "foo"?

Anyways, all this said - I'm willing to let this behavior stand for the initial implementation. We can let the implementation evolve with more usage.